adobe/forest.hpp

```diff
@@ -260,14 +253,16 @@ class reverse_fullorder_iterator
           std::bidirectional_iterator_tag, typename boost::iterator_reference<I>::type> {
     typedef typename boost::iterator_facade<
         reverse_fullorder_iterator<I>, typename boost::iterator_value<I>::type,
-        std::bidirectional_iterator_tag, typename boost::iterator_reference<I>::type> inherited_t;
+        std::bidirectional_iterator_tag, typename boost::iterator_reference<I>::type>
+        inherited_t;
 
 public:
     typedef I iterator_type;
     typedef typename inherited_t::reference reference;
 
     reverse_fullorder_iterator() : edge_m(forest_trailing_edge) {}
-    explicit reverse_fullorder_iterator(I x) : base_m(--x), edge_m(forest_leading_edge - base_m.edge()) {}
+    explicit reverse_fullorder_iterator(I x)
+        : base_m(--x), edge_m(forest_leading_edge - base_m.edge()) {}
     template <typename U>
     reverse_fullorder_iterator(const reverse_fullorder_iterator<U>& x)
         : base_m(--x.base()), edge_m(forest_leading_edge - base_m.edge()) {}
```

Comment:
I think this is a bug; no need to decrement the iterator when it's already a `reverse_fullorder_iterator`
```suggestion
        : base_m(x.base()), edge_m(forest_leading_edge - base_m.edge()) {}
```

Reply:
Should we simply replace adobe::forest with:
```
namespace adobe {

using stlab::forest;
// and other associated types

}
```

That's how I handled the migration of `adobe::copy_on_write` -> `stlab::copy_on_write`

Reply:
Also - are you sure about the fix? `I` is a `fullorder_iterator`. I think you intended to change the next case.

Reply:
Yes, you're right. I intended to change the next case, not the one in the suggestion.

Reply:
I'm fine with ASL having a dependency on stlab; that seems like a pretty big change to me, though, but if we're already doing it for copy_on_write, then I think it should be the way to go in this case, too.

Reply:
As for the change, I think it's actually on the correct line, even though GitHub is highlighting another one. The suggestion shows line 269, which is the `reverse_fullorder_iterator` case.